sway-core/src/type_system/mod.rs

```diff
@@ -7,7 +7,7 @@ mod resolved_type;
 mod trait_constraint;
 mod type_argument;
 mod type_binding;
-mod type_engine;
+pub(crate) mod type_engine;
```

Comment:
Instead of changing the visibility of the module, change the import statement to match this example: https://github.com/FuelLabs/sway/blob/9076eb232cb24d5ed9136d7e8d7a77d4f7460467/sway-core/src/semantic_analysis/type_check_context.rs#L6